### DISCUSSION TITLE
OneDrive transfer fiasco

### DISCUSSION THREAD

**Discussion by adgbu** [iterate-ch/cyberduck#16598] [Q&A]
TL;DR: Mountain Duck was tasked to sync (download) a folder on my OneDrive that contains 22 GB of photos which is about 220 000 files. One week later Mountain Duck has not beed able to finish this task. The UI is unresponsive. There is also no way for me to cancel the operation and give up. What am I supposed to do?

I am using Mountain Duck on macOS (Sonoma) to mount my personal OneDrive and the company's SharePoint storage. This normally works fine, albeit generally slow in transfers which might be the fault of Microsoft. 

Now I needed to access a set of photos a folder on my OneDrive which is offloaded to the cloud and not available locally on my computer. In the Finder context menu under Mountain Dock I marked the folder to "keep offline on local disc". After that Mountain Duck started to transfer files. The Mountain Duck UI in the menu bar says "Syncronization ongoing, 220 000 files, 21,7 GB".

From this point on the Mountain Duck UI is completely unresponsive. The cursor is the spinning beach ball. The menu bar menu/window never closes and obscures the top right corner of my screen forever. In Activity Monitor, Mountain Duck is listed as "not responding", CPU usage is 250-300%, memory usage is > 4 GB, energy usage is > 5000 (200x more than any other app). The computer runs hot. It is still possible to terminate the Mountain Duck app without forcing it to be killed.

As soon as I start the Mountain Duck app again the same thing happens. Unresponsive UI, high resource usage, little or no progress on the transfer. I left it to work for several days like this and after 5 days it had progressed to 160 000 files or 15 GB remaining. Now a few days later the progress has reverted back to the beginning with 220 000 files remaining.

It is clear to me that this will never work. Mountain Duck cannot handle this scenario. All I want to do now is give up and only use Mountain Duck, OneDrive and SharePoint for trivial workloads like a handful of spreadsheets etc. But there is no way for me to cancel this transfer operation. How can I make Mountain Duck give up on the transfer? Until it does, it blocks all usage of OneDrive/SharePoint and sucks all the resources out of my computer.

**Comment by dkocher** [comment]
I understand that you selected _Keep Offline_. To reverse and abort the downloads in progress, try the following:

1. Choose _Pause Synchronization_ in the Mountain Duck Sync menu.
2. Select _Delete on local Disk_ within the Mountain Duck [context menu](https://docs.mountainduck.io/mountainduck/interface/#context-menu-in-finder-and-windows-file-explorer).
3. _Resume Synchronization_ in the Mountain Duck Sync menu.

**Reply by adgbu** [reply]
I would love to try that. The problem is that the UI is completely frozen, giving me only the spinning beach ball. I cannot select anything in the Mountain Duck menu. This repeats every time I restart Mountain Duck or reboot the computer.

**Reply by tatze96** [reply]
You can follow the steps below to clear the synchronization queue if the steps above, contrary to expectations, don't work.
1. Disconnect all bookmarks but the affected bookmark.
2. Navigate into the _Sessions_ folder by choosing `Go → Go to folder` within the _Finder_ menu, copying the path below into the appearing window, and clicking on the _Open_ button afterward.
`~/Library/Group Containers/G69SCX94XU.duck/Library/Application Support/duck/Sessions/`
3. Note down the file name of the currently connected bookmark within the _Sessions_ folder, for example, within a text file.
4. Quit Mountain Duck.
5. Navigate into the _Sync_ folder by following step 2 but with the path below.
`~/Library/Group Containers/G69SCX94XU.duck/Library/Application Support/duck/Sync/`
6. Delete the folder corresponding to the previously noted-down file name.
7. Start Mountain Duck.

Note that the items previously located within the synchronization queue won't be added to the queue again until you edit, re-synchronize, or delete them again.

**Comment by adgbu** [comment]
The answer from @tatze96 is probably good. I solved it another way before I saw that answer.

What I ended up doing was this:
1. Disconnected the computer from Internet (both WiFi and Ethernet).
2. Without Internet I could open Mountain Duck UI and it was responsive.
3. Removed the bookmark for OneDrive, which had the big transfer.
4. Quit Mountain Duck.
5. Reconnected to the netowrk.
6. Restarted Mountain Duck.
7. Created a new bookmark for OneDrive. Apparently this created a new entry in the Sync folder. 
```~/Library/Group Containers/G69SCX94XU.duck/Library/Application Support/duck/Sync/```
Sync back to normal. The transfer cancelled.
